### ISSUE TITLE
Possibility of using conditional speed limit?

### ISSUE THREAD

**Issue by AravindaDP** [pfeiferj/mapd#50]
OpenStreetMap allows defining conditional speed limits. https://wiki.openstreetmap.org/wiki/Key:maxspeed:conditional

Is it possible to add this to Ways returned? https://github.com/pfeiferj/mapd/blob/main/generate_offline.go#L150 And then in https://github.com/pfeiferj/mapd/blob/main/mapd.go#L179 can we alter the `MAP_SPEED_LIMIT` based on current time and if conditional speed applies. Also `NEXT_MAP_SPEED_LIMIT` can be adjusted based on current time as well (For the brevity omitting the complexity of considering what will be the time when we reach next way) https://github.com/pfeiferj/mapd/blob/main/mapd.go#L249

This would make it necessary to mapsd to read current local time which I assume is a fair dependency.

If contributions are accepted, I can give it a try.

**Comment by pfeiferj** [comment]
@AravindaDP yeah, feel free to create something, but throw it behind an option in a param so it can be enabled or disabled

**Comment by dirkpetersen** [comment]
@pfeiferj I could pick this up, if you're still open to it. With #89 merged you presumably have a planet regeneration coming — and since this feature also needs a new field in the offline way data, it seems like the right moment to land the schema part so both changes ride the same rebuild instead of needing another regeneration later?

A few scoping questions before I write anything:

1. Scope: your roadmap lists this as "limited support… only simple conditions that are parseable and time based". Still the scope you want — day/time conditions like 25 mph @ (Mo-Fr 07:00-17:00), with weather/vehicle-class/wet conditions ignored entirely?
2. Storage shape: would you prefer storing the raw maxspeed:conditional tag text on the way, or parsing it at generation time into a compact structure? I lean raw text: only the small fraction of ways carrying the tag pay any size cost, and the runtime parser can then improve (or fix bugs) in later binary releases without another tile regeneration — but it's your format, so your call.
3. The option: your original comment said to put it behind a param — in v2 terms I assume that means an opt-in setting in defaults.json (off by default) with a matching MapdInputType and settings.Handle case, like the other toggles?

**Comment by pfeiferj** [comment]
Yeah, definitely open to PRs for this

> A few scoping questions before I write anything:
> 
> 1. Scope: your roadmap lists this as "limited support… only simple conditions that are parseable and time based". Still the scope you want — day/time conditions like 25 mph @ (Mo-Fr 07:00-17:00), with weather/vehicle-class/wet conditions ignored entirely?

From a behavior standpoint the scope would just be that speed control inside mapd would only handle the day/time stuff. It's fine to store the other things and to output the raw values for forks to use.

> 2. Storage shape: would you prefer storing the raw maxspeed:conditional tag text on the way, or parsing it at generation time into a compact structure? I lean raw text: only the small fraction of ways carrying the tag pay any size cost, and the runtime parser can then improve (or fix bugs) in later binary releases without another tile regeneration — but it's your format, so your call.

Yeah, raw values stored as strings would be fine. Like you said, there isn't that many of these tags in osm and the text is going to have a good compression ratio anyways

> 3. The option: your original comment said to put it behind a param — in v2 terms I assume that means an opt-in setting in defaults.json (off by default) with a matching MapdInputType and settings.Handle case, like the other toggles?

Yeah, it would just be like the other toggles. Defaulted to off just because it's a little bit more "advanced" behavior that can be harder to communicate to users

**Comment by dirkpetersen** [comment]
PR is up: #94 — implemented along the lines above (raw text stored, day/time-only evaluation behind a default-off toggle, raw tag output for forks). Since it adds Way fields, it would ideally ride the same regeneration as the highwayClass field if the timing works out.
